File: gestion/src/main/java/com/hps/gestion/dto/Commit.java
package com.hps.gestion.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Commit {
    private String hash;
    private String message;
    private LocalDateTime date;
    private String author;
    private String repository;
    private String project;
    private String branch;
    private String jiraID;
    private String type;
    private String issueID;
    private String summary;
}

File: gestionUI/src/app/shared/commit.ts
import { Ticket } from "./ticket";

export class Commit{
    hash : string;
    message : string;
    date : string;
    author : string;
    repository : string;
    project : string;
    branch : string;
    jiraID : string;
    type : string;
    issueID : string;
    summary : string;

    constructor(hash: string, message: string, date: string , author : string, repository : string, project : string ,branch : string , jiraID : string, type : string, issueID : string , summary : string) {
        this.hash = hash;
        this.message = message;
        this.date = date;
        this.author = author;
        this.repository = repository;
        this.project = project;
        this.branch = branch;
        this.jiraID = jiraID;
        this.type = type;
        this.issueID = issueID;
        this.summary = summary;

    }
}